src/main/java/lambdaperson/FilteredPersons.java

```diff
@@ -0,0 +1,23 @@
+package lambdaperson;
+
+import personlist.Person;
+import java.util.Comparator;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class FilteredPersons {
+
+    public static Set<Person> filterPersons(Set<Person> personList) {
+        Comparator<Person> sortingByNameLength =
+                Comparator.comparingInt((Person p) -> p.getName().length());
+
+        return personList.stream().filter(Person -> Person.getAge() < 30 & Person.getAge() > 20)
```

Comment:
It's not a good practice to use `&` operator for such cases. `&&` is better.
https://stackoverflow.com/questions/5564410/what-is-the-difference-between-and-in-java